src/parsing/compiledExpressionCache.ts

```diff
@@ -102,6 +106,7 @@ export function getStaticCompilationResultFromCache(
 
 	const cacheWithCorrectContext = cachesForLanguage.find(
 		cache =>
+			cache.debug === debug &&
```

Comment:
This check ensures that the static evaluation is performed again for the expression. However this will not add or remove the `StackTraceGenerator`s from the expression.

So lets say we execute the same expression twice both with `disableCache: false`. But the first with `debug: true` and the second with `debug: false`. In this case the second one will also have the `StackTraceGenerator`s.

Wouldn't it make more sense to treat the debug flag the same as the language?